students/YingGuo/session04/Trigrams.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+"""Session 04, assignment:Trigrams, student name: Ying Guo"""
+
+words = "I wish I may I wish I might good morning good afternoon hahahahaha lala hahaha".split()
+
+from collections import defaultdict
+import random
+
+def build_trigrams(words):
+
+    trigrams = defaultdict(list)
+
+    for i in range(len(words)-2):
+        pair = tuple(words[i:i+2])
+        follower = words[i+2]
+        trigrams[pair] += [follower]
+    
+    print("This is the Trigrams:")
+    for i in trigrams.keys():
+        print(i, ": ", trigrams[i])
+
+    return trigrams
+        
+def build_text(trigram_dict):
+    key_lst = list(trigram_dict.keys())
+    print("This is the Tuple key list:\n", key_lst)
+    fst_item = random.choice(key_lst)
+    new_lst = [fst_item[0].capitalize(),fst_item[1]]
+    while len(new_lst) <=200:
+        key_tuple = (new_lst[-2],new_lst[-1])
+        if key_tuple in key_lst:
```

Comment:
Minor again: No need to create a separate list `key_lst` for this purpose. You could have just used `if key_tuple in trigram_dict:`, because `k in a_dict` is about whether the key `k` is in the `a_dict` or not.